src/python/pants/pantsd/watchman.py

```diff
@@ -137,7 +142,12 @@ def watch_project(self, path):
 
     :param string path: the path to the watchman project root/pants build root.
     """
-    return self.client.query('watch-project', os.path.realpath(path))
+    # TODO(kwlzn): Add a client.query(timeout=X) param to the upstream pywatchman project.
```

Comment:
in the general context, it could be - yeah. in our current usage case, it isn't ever - we only ever watch the singular build root once per daemon launch.

I had gone the route you suggested initially, but due to issues in the pywatchman upstream client calling `client.setTimeout()` call before issuing an actual query (which is the only way to invoke the private `client._connect()` without violating the private method convention) fails with an `AttributeError` trying to set an attribute on a `None`. we could attempt that each time in a try/except block and catch `AttributeError` - such that any subsequent secondary calls to `.watch_project()` respect the startup_timeout - but ultimately based on the way we use this class it it's not likely to affect us afaict, so just went with the simplest strategy w/ a TODO. would be happy to do the try/except if you think that sounds better - as it is more correct for the general case, but imo mildly kludgy.

definitely hope to circle back to improve the upstream pywatchman client a bit and clean this and at least one other TODO up.